File: EnglishLearning.Domain/Entities/Quiz.cs
using EnglishLearning.Domain.Common;
using EnglishLearning.Domain.Enums;

namespace EnglishLearning.Domain.Entities;

public class Quiz : BaseEntity
{
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DifficultyLevel Difficulty { get; set; }
    public int TimeLimitMinutes { get; set; }
    public decimal PassingScore { get; set; } = 50m;
    public DateTime? StartTime { get; set; }
    public DateTime? EndTime { get; set; }

    // Navigation
    public ICollection<Question> Questions { get; set; } = new List<Question>();
    public ICollection<QuizResult> Results { get; set; } = new List<QuizResult>();
}

File: EnglishLearning.Client/src/types/quiz.ts
import { DifficultyLevel } from './vocabulary'
import { UserRole } from './auth'

export interface Quiz {
  id: string
  title: string
  description?: string
  difficulty: DifficultyLevel
  timeLimitMinutes?: number
  passingScore?: number
  createdAt: string
}

export interface Question {
  id: string
  questionText: string
  questionType: QuestionType
  choices: ChoiceForTake[]
}

export interface ChoiceForTake {
  id: string
  choiceText: string
}

export interface Choice extends ChoiceForTake {
  isCorrect: boolean
}

export type QuestionType = 'MultipleChoice' | 'TrueFalse' | 'FillInBlank'

export interface QuizAssignment {
  id: string
  quizId: string
  quizTitle: string
  targetRole?: UserRole
  targetUserId?: string
  startTime: string
  endTime: string
  status: AssignmentStatus
}

export type AssignmentStatus = 'Scheduled' | 'Active' | 'Completed' | 'Cancelled'

export interface QuizResult {
  id: string
  quizId: string
  userId: string
  score: number
  totalQuestions: number
  correctAnswers: number
  submittedAt: string
  answers: QuizAnswer[]
}

export interface QuizAnswer {
  questionId: string
  selectedChoiceId?: string
  answerText?: string
}
